lib/abci/handlers/endBlockHandlerFactory.js

```diff
@@ -0,0 +1,52 @@
+const {
+  abci: {
+    ResponseEndBlock,
+  },
+} = require('abci/types');
+
+const NoDPNSContractFoundError = require('../errors/NoDPNSContractFoundError');
+
+/**
+ * Begin block ABCI handler
+ *
+ * @param {BlockchainState} blockchainState
+ * @param {BlockExecutionDBTransactions} blockExecutionDBTransactions
+ * @param {BlockExecutionState} blockExecutionState
+ * @param {Number} protocolVersion - Protocol version
+ * @param {BaseLogger} logger
+ *
+ * @return {endBlockHandler}
+ */
+function endBlockHandlerFactory(
+  blockchainState,
+  blockExecutionDBTransactions,
+  blockExecutionState,
+  protocolVersion,
+  logger,
+) {
+  /**
+   * @typedef endBlockHandler
+   *
+   * @param {abci.RequestBeginBlock} request
+   * @return {Promise<abci.ResponseBeginBlock>}
+   */
+  async function endBlockHandler({ header }) {
+    logger.info(`Block end #${header.height}`);
+
+    const DPNSEnvsAreDefined = DPNS_CONTRACT_HEIGHT && DPNS_CONTRACT_ID;
```

Comment:
We usually name such variables like "isEnable" for example. so it should be `areDPNSEnvsDefined`?

Reply:
Also, they should come from DI container.